File: src/app/language_profile.ts
/**
 * Minimal centralized language profile (v1).
 *
 * RU-first by design. No runtime language switching is exposed to the user yet.
 * All runtime language decisions flow through this module so there is a single
 * source of truth instead of scattered `"ru"` literals.
 */

export const DEFAULT_LANGUAGE = "ru";
export type LanguageCode = "ru" | "en";

export interface LanguageProfile {
  code: LanguageCode;
}

/**
 * Returns the current application language code.
 * Currently always RU — future versions may read from settings or OS locale.
 */
export function currentLanguage(): LanguageCode {
  return DEFAULT_LANGUAGE;
}

/**
 * Returns the current language profile.
 * Currently always RU profile — future versions may read from settings or OS locale.
 */
export function currentProfile(): LanguageProfile {
  return { code: DEFAULT_LANGUAGE };
}

File: src-tauri/src/language_profile.rs
//! Minimal centralized language profile (v1).
//!
//! RU-first by design. No runtime language switching is exposed to the user yet.
//! All runtime language decisions flow through this module so there is a single
//! source of truth instead of scattered `"ru"` literals.

/// Default application language — currently always "ru".
/// Future versions may read from settings or OS locale.
pub fn default_language() -> &'static str {
    "ru"
}

/// STT (Deepgram) language code for the default profile.
pub fn stt_language() -> &'static str {
    "ru"
}

/// LLM prompt language code for the default profile.
pub fn llm_language() -> &'static str {
    "ru"
}
